### ISSUE TITLE
RequestLogger doesn't log GET requests when logBody is set

### ISSUE THREAD

**Issue by JanGe** [http4s/http4s#1849]
I'm wrapping my service with the `Logger` middleware and `logBody` set to `true`:
```scala
Logger(logHeaders = true, logBody = true)(service)
```

Unfortunately, the `RequestLogger` is not logging GET requests:
```
18-05-10T15:14:27.684 INFO  run-main-0 o.h.s.b.BlazeBuilder - http4s v0.18.10 on blaze v0.12.13 started at http://[0:0:0:0:0:0:0:0]:9630/
18-05-10T15:15:31.417 INFO  global-212 o.h.s.m.ResponseLogger - HTTP/1.1 200 OK Headers(Content-Type: application/json, Content-Length: 2, X-Request-Duration: 791ms) body="[]"
```

Once I turn `logBody` to `false`, it is logging the request just fine:
```
18-05-10T15:18:29.765 INFO  run-main-0 o.h.s.b.BlazeBuilder - http4s v0.18.10 on blaze v0.12.13 started at http://[0:0:0:0:0:0:0:0]:9630/
18-05-10T15:18:34.750 INFO  -global-93 o.h.s.m.RequestLogger - HTTP/1.1 GET /health Headers(Accept: */*, Host: localhost:9630, cookie: <REDACTED>, accept-encoding: gzip, deflate, Connection: keep-alive)
18-05-10T15:18:35.140 INFO  -global-93 o.h.s.m.ResponseLogger - HTTP/1.1 200 OK Headers(Content-Type: application/json, Content-Length: 2, X-Request-Duration: 463ms)
```

**Comment by rossabaker** [comment]
This looks a lot like what just got fixed in #1846 on the client side.

**Comment by rossabaker** [comment]
I think we need to put the `Logger.logMessage` call in the finalizer of the response body (which is guaranteed to be consumed), not the finalizer of the request body (which isn't guaranteed to run -- and typically wouldn't be on bodyless methods like `GET`).
